trie.h

```diff
@@ -0,0 +1,37 @@
+#ifndef TRIE_H
+#define TRIE_H
+
+#include <map>
+#include <vector>
+#include <string>
+#include <memory>
+
+using std::string;
+using std::map;
+using std::vector;
+using std::shared_ptr;
+
+class Trie {
+private:
+    struct Node {
+        map<char, shared_ptr<Node>> children;
```

Comment:
Did you not put a space between the '>>' on purpose just to see if the compiler really could handle it now? That is one of the features of C++11, I think.